README.md

```diff
@@ -97,6 +97,24 @@ Acquires the semaphore asynchronously. The wait operation is performed on a sepa
 
 Acquires the semaphore synchronously. The wait operation is performed on the main thread, and will block the event loop until the semaphore is acquired. Throws if an error occurs.
 
+### `Semafour.prototype.tryWait(callback)`
+
+  - Arguments
+    - `callback(err, succeeded)` (function) - Callback function, which takes an error argument, and a boolean indicating whether the semaphore was locked.
+  - Returns
+    - Nothing
+
+Tries to acquire the semaphore asynchronously. The tryWait operation is performed on a separate thread in the libuv thread pool to avoid blocking the event loop. Once the semaphore is acquired (if possible), the callback is invoked. Any errors that occur are passed to the callback function.
```

Comment:
Does it make sense to use the libuv threadpool here? Doesn't `sem_trywait()` return immediately?

Reply:
Probably not for a semaphore lock. I've seen non-blocking socket reads take time (for reasons I can't remember), and I just assumed it'd be the same for `sem_trywait`.

I'll try to research this a bit before implementing a change.

Reply:
My (possibly erroneous) feeling is that It's kind of a Node.js convention to do sync and async versions of every single I/O function (I'm thinking of the `fs` module in particular).

Is it true or am I mistaken?

Reply:
The Node convention is to avoid blocking operations because JavaScript executes in a single thread. That said, Node is full of synchronous operations (see the `buffer` module for example) where it makes sense.

If `sem_trywait()` returns immediately, then it doesn't make sense to ship it to the thread pool.

As a side note, I believe the `fs.*Sync()` functions are more about providing convenience for things like shell scripts where blocking the event loop doesn't matter. It's also a bit of an implementation detail, because libuv has to implement the fs operations synchronously, and just provides the illusion of async by running the synchronous code in the thread pool. There aren't, for example, synchronous network requests.

Reply:
... and code will be vastly improved by removing the async version anyway 👍